### PR TITLE
Add CE Secure Marker

### PR DESCRIPTION
**Pull request by LeoKHoward**
# Motivation and Context
New column for identifying if CE case is a Secure Establishment

# What has changed
New ceSecure/secureEstablishment column 
Tests

# How to test?
Clone the branch and run ```maven clean install``` which will build the image

# Links
https://collaborate2.ons.gov.uk/confluence/display/CRM/Address+Frame+specification+2021?focusedCommentId=36183836#comment-36183836
https://trello.com/c/BAI9qOOC/619-rmc-341-add-secure-marker-case-api-and-field-8

### CODE REVIEW

**Comment by LeoKHoward** [comment]
[RMC-341 Add secure marker - case api and field (8)](https://trello.com/c/BAI9qOOC/619-rmc-341-add-secure-marker-case-api-and-field-8)

File: src/main/java/uk/gov/ons/census/caseapisvc/endpoint/CaseEndpoint.java [hunk 1]
```diff
@@ -201,6 +201,10 @@ private CaseContainerDTO buildCaseContainerDTO(Case caze, boolean includeCaseEve
     CaseContainerDTO caseContainerDTO = this.mapperFacade.map(caze, CaseContainerDTO.class);
     caseContainerDTO.setSurveyType(caze.getSurvey());
 
+    //    if (caze.getMetadata() != null) {
+    caseContainerDTO.setCeSecure(caze.getMetadata().getSecureEstablishment());
+    //    }
```
Comment (AdamHawtin) [inline src/main/java/uk/gov/ons/census/caseapisvc/endpoint/CaseEndpoint.java]: Commented out null check should either be deleted or reinstated

**Review by AdamHawtin — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/ons/census/caseapisvc/endpoint/CaseEndpoint.java (hunk 1 above)
Comment (LukeLoze1) [reply src/main/java/uk/gov/ons/census/caseapisvc/endpoint/CaseEndpoint.java]: I'm presuming if this passes the ATs with it commented out then it's 'probably' not needed

File: src/test/java/uk/gov/ons/census/caseapisvc/endpoint/CaseEndpointIT.java [hunk 2]
```diff
@@ -887,6 +883,46 @@ public void testFindCcsCaseByPostCodeIgnoresCharacterCase() throws IOException,
     assertThat(foundCases.get(0).getCaseId()).isEqualTo(ccsCase.getCaseId().toString());
   }
 
+  @Test
```
Comment (LukeLoze1) [inline src/test/java/uk/gov/ons/census/caseapisvc/endpoint/CaseEndpointIT.java:886]: Not super keen on having 2 new Integration Tests just to check a field is being set correctly. Won't block the ticket on it.  I'd like to see that Endpoint and it's tests refactored, there's 1083 lines in the CaseEndpointITs, this seems mad, mad I tell you.

**Review by LukeLoze1 — COMMENTED** [review]
Remove commented out code

File: src/main/java/uk/gov/ons/census/caseapisvc/endpoint/CaseEndpoint.java (hunk 1 above)
Comment (NickGrantX) [reply src/main/java/uk/gov/ons/census/caseapisvc/endpoint/CaseEndpoint.java]: yes the code that's commented out needs to be deleted

File: src/test/java/uk/gov/ons/census/caseapisvc/endpoint/CaseEndpointIT.java (hunk 2 above)
Comment (NickGrantX) [reply src/test/java/uk/gov/ons/census/caseapisvc/endpoint/CaseEndpointIT.java:886]: at least they run quite quickly. Each test is also quite short. It's only test code and it has good coverage, and it's easy to edit if a test fails.

Comment (NickGrantX) [reply src/test/java/uk/gov/ons/census/caseapisvc/endpoint/CaseEndpointIT.java:886]: The problem is that we're starting to get "God Classes" that do everything including the kitchen sink. Combining telephone capture, fulfilments, plus general CRUD crap all into one place is the real problem here.

**Review by NickGrantX — APPROVED** [review]
LGTM. Ignore the haters.

Haters gonna hate.

File: src/main/java/uk/gov/ons/census/caseapisvc/endpoint/CaseEndpoint.java (hunk 1 above)
Comment (AdamHawtin) [reply src/main/java/uk/gov/ons/census/caseapisvc/endpoint/CaseEndpoint.java]: The lack of null check here seems to be causing the AT failures, for CCS cases it's throwing NPE's. Either needs to be null checked here or ensure we always create an (empty) metadata object on cases even through the CCS channels.

**Review by AdamHawtin — DISMISSED** [review]

File: src/main/java/uk/gov/ons/census/caseapisvc/endpoint/CaseEndpoint.java (hunk 1 above)
Comment (NickGrantX) [reply src/main/java/uk/gov/ons/census/caseapisvc/endpoint/CaseEndpoint.java]: hmmm good point. The null check isn't going to hurt. Let's just put it back in - that's the quickest and easiest.

Comment (LeoKHoward) [reply src/main/java/uk/gov/ons/census/caseapisvc/endpoint/CaseEndpoint.java]: Have put it back in and the CCS tests now pass so will be keeping it

File: src/test/java/uk/gov/ons/census/caseapisvc/endpoint/CaseEndpointIT.java (hunk 2 above)
Comment (LukeLoze1) [reply src/test/java/uk/gov/ons/census/caseapisvc/endpoint/CaseEndpointIT.java:886]: I like the term 'God Classes'

**Review by LukeLoze1 — APPROVED** [review]

**Pull request merged**